keps/sig-api-machinery/1965-kube-apiserver-identity/README.md

```diff
@@ -65,14 +80,24 @@ load balancer for the cluster, where the advertise IP address is set to the IP
 address of the load balancer, all three kube-apiservers will have the same
 advertise IP address.
 
+### Goals
+
+* Provide a mechanism in which controllers can uniquely identify kube-apiserver's in a cluster.
+
+### Non-Goals
+
+* improving the availability of kube-apiserver
+
 ## Proposal
 
 We will use “hostname+PID+random suffix (e.g. 6 base58 digits)” as the ID.
```

Comment:
Are all e.g. cloud providers actually OK with letting the hostname of systems running apiserver be known?

My first idea: just use a UUID. Also add a "nickname" (label on the lease object?), something persisted to disk that is the same across restarts of the "same" apiserver, to help humans understand continuity.

Reply:
> Are all e.g. cloud providers actually OK with letting the hostname of systems running apiserver be known?

I think most cloud providers work around this by just overriding the hostname or using an opaque value that doesn't reveal anything about the system. We also use hostnames for kube-controller-manager/kube-scheduler leases
already right?

> My first idea: just use a UUID. Also add a "nickname" (label on the lease object?), something persisted to disk that is the same across restarts of the "same" apiserver, to help humans understand continuity.

Fwiw, seems like we are already using UUID: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/config.go#L326, so hostname for apiserver identity specifically doesn't seem like an issue in the current form. I like the label idea but will give it more thought.

Reply:
> My first idea: just use a UUID. Also add a "nickname" (label on the lease object?), something persisted to disk that is the same across restarts of the "same" apiserver, to help humans understand continuity.

I'm ok with this as long as its also easy for someone to run as a pod in the cluster.  I'm also supportive of hashing whatever name we choose